pandas/core/dtypes/cast.py

```diff
@@ -1026,3 +1027,52 @@ def find_common_type(types):
             return np.object
 
     return np.find_common_type(types, [])
+
+
+def maybe_cast_to_integer(arr, dtype):
+    """
```

Comment:
This actually can take *any* dtype and will return a casted version. It happens to check for integer/unsigned integer dtypes. So pls expand a little.